source/manuals/programming-languages/python/linting.html.md.erb

```diff
@@ -0,0 +1,150 @@
+---
+title: Linting Python at GDS
+expires: living
+---
+
+# <%= current_page.data.title %>
+
+
+### Flake8
+
+This manual advises the use of the [Flake8][Flake8] all in one lint, codestyle and complexity checker.
+
+
+### What is Flake8?
+
+[Flake8][Flake8] is a command line utility that acts as a drop in replacement for the [pep8/ PyCodeStyle][pycodestyle]command line checker.
+If you're already using the [pep8/ PyCodeStyle][pycodestyle]checker you're most of the way there. If not then, never fear. You'll find everything
+you need on this page.
+
+[Flake8][Flake8] is effectively a wrapper around 3 things:
+* [pep8/ PyCodeStyle][pycodestyle]checks that your code conforms to [PEP-0008][PEP8]
+* [McCabe complexity checker][McCabe] checks the [complexity][WikiCyclomatic_complexity] of your code against a configurable maximum complexity level
+* [PyFlakes][PyFlakes] checks for 'suspicious constructs' like unused imports, unused variable declarations and code paths that can never execute
+
+These tools together help developers write readable and maintainable code and catch bugs.
+
+
+### How to use Flake8
+
+Implementation of [Flake8][Flake8] will depend on whether the repository you want to run the checks on is a module or an application,
+and how your dependencies and automated test ci is set up.
+
+You'll want to add the [Flake8][Flake8] module (available from [PyPI][PYPI]) to your 'dev' or 'test' requirements/ dependencies.
+You'll then likely want to run it before you run your unit-atests.
+
+### Example usage
+
+For a quick example run the below code (assuming you have [`virtualenv`][virtualenv] installed)
+
+```bash
+mkdir flake8_test; echo 'import os' > flake8_test/flake8_test_file.py
+virtualenv flake8_test/venv
```

Comment:
If we're saying use Python 3, I'd suggest we make sure Python3 is used in examples?

`python3 -m venv flake8_test/venv`

Reply:
Done 👍